### ISSUE TITLE
New Textobj: chainmember

### ISSUE THREAD

**Issue by ofirgall** [chrisgrieser/nvim-various-textobjs#26]
### Textobj requested

Textobject of a chain member in a call line.
Explained better here: https://github.com/D4KU/vim-textobj-chainmember

### Inner/Outer Difference

outer:
```
cursor position │ .....
buffer line     │ first.second.third
selection       │ └ am ┘

Another examples:
     .......
first.second.third
     └ am ─┘

     ........               .
first.second(arg1.var, foo()).third
     └───────── am ─────────┘
```

Inner (without the dots):
```
     .......
first.second.third
      └ im ┘
```

### Vimscript plugin already implementing the textobj

https://github.com/D4KU/vim-textobj-chainmember

### Checklist

- [X] The textobj would be useful to more users than just me.
- [X] There is no lua plugin already implementing the textobj in an adequate manner.
- [X] I read the docs and there is no such textobj already.

**Comment by chrisgrieser** [comment]
Thanks for the nice explanation how the text object is supposed to work!

I am wondering, wouldn't treesitter-textobjs' outer call object cover most of the cases for `am`? https://github.com/nvim-treesitter/nvim-treesitter-textobjects#built-in-textobjects

**Comment by ofirgall** [comment]
Unfortunately outer call object doesn't cover fields, only calls

**Comment by chrisgrieser** [comment]
I see, yeah that makes sense.

Hmm, this plugin implements most text objects by a form of pattern matching. Especially with the potential for nexted fields, I think this might be tricky to implement via pattern matching. 🤔 

I'm gonna have to think about it, but I think it might make more sense to implement this as a [treesitter-based textobj](treesitter-based textobj)

**Comment by ofirgall** [comment]
Okay, I hoped I could get rid of another VimL plugin.
Btw thanks for the plugin, I didn't know I need those textobjects :)

**Comment by chrisgrieser** [comment]
actually, I learned that lua patterns have the quite handy pattern token `%bxy`, which actually make it super easy to handle nested calls. So I went ahead and added chainMember as a new textobj. 

Do note though that since lua patterns do not know logical ors `|`, the chainMember textobj *has to* include a call, chain member without call are not recognized. (Though chainmembers without call can probably be handled as `a.` and `i.` via other plugins already.)
